public/app/features/dashboard/state/DashboardModel.ts

```diff
@@ -167,7 +174,16 @@ export class DashboardModel {
 
     this.addBuiltInAnnotationQuery();
     this.sortPanelsByGridPos();
-    this.hasChangesThatAffectsAllPanels = false;
+    this.panelsAffectedByVariableChange = null;
+    this.appEventsSubscription = new Subscription();
+    this.lastRefresh = Date.now();
+    this.appEventsSubscription.add(appEvents.subscribe(VariablesChanged, this.variablesChangedHandler.bind(this)));
+    this.appEventsSubscription.add(
+      appEvents.subscribe(VariablesTimeRangeChanged, this.variablesTimeRangeChangedHandler.bind(this))
+    );
+    this.appEventsSubscription.add(
+      appEvents.subscribe(VariablesChangedInUrl, this.variablesChangedInUrlHandler.bind(this))
+    );
```

Comment:
To support nested dashboards (with their own variables), using appEvents here (vs dashboard events) could be a problem in terms of scope. But very much depends on the future of DashboardModel and dashboard.events and how likely we will use this event system in the future. And it seems easy to switch to dashboard.events (if we need to), the flip side is if we use dashboard.events we link variable system to dashboard more and make them less standalone.

Reply:
Maybe we can revisit that when we know more about nested dashboards.